pytest_fixtures/core/contenthosts.py

```diff
@@ -19,20 +19,21 @@ def host_conf(request):
         params = request.param
     distro = params.get('distro', 'rhel')
     _rhelver = f"{distro}{params.get('rhel_version', settings.content_host.default_rhel_version)}"
+    # check to see if no-containers is passed as an argument to pytest
+    deploy_kwargs = {}
+    if not request.config.getoption('no-containers', None) and params.get('use_containers', True):
+        deploy_kwargs = settings.content_host.get(_rhelver).get('container', {})
+    # if we're not using containers or a container isn't available, use a VM
+    if not deploy_kwargs:
+        deploy_kwargs = settings.content_host.get(_rhelver).get('vm', {})
+    conf.update(deploy_kwargs)
+    # CI-defined params specific to only a few jobs
     rhel_compose_id = settings.get(f"content_host.deploy_kwargs.{_rhelver}.compose")
     if rhel_compose_id:
         conf['deploy_rhel_compose_id'] = rhel_compose_id
     scenario = settings.get(f"content_host.deploy_kwargs.{_rhelver}.scenario")
     if scenario:
         conf['deploy_scenario'] = scenario
```

Comment:
@rplevka @ogajduse i'd like to consult the both of you on how to handle this better